components/user_settings/display/manage_languages/manage_languages.tsx

```diff
@@ -1,144 +1,176 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
 import React from 'react';
 import {FormattedMessage} from 'react-intl';
 import ReactSelect from 'react-select';
+import {ActionResult} from 'mattermost-redux/types/actions';
+import {UserProfile} from 'mattermost-redux/types/users';
 
 import * as I18n from 'i18n/i18n.jsx';
 import SettingItemMax from 'components/setting_item_max.jsx';
 import FormattedMarkdownMessage from 'components/formatted_markdown_message.jsx';
 import {isKeyPressed} from 'utils/utils.jsx';
 import Constants from 'utils/constants';
 
-export default class ManageLanguage extends React.PureComponent {
-    static propTypes = {
-        user: PropTypes.object.isRequired,
-        locale: PropTypes.string.isRequired,
-        updateSection: PropTypes.func.isRequired,
-        actions: PropTypes.shape({
-            updateMe: PropTypes.func.isRequired,
-        }).isRequired,
-    };
-
-    constructor(props) {
+type Actions = {
+    updateMe: (user: UserProfile) => Promise<ActionResult>;
+};
+
+type Props = {
+    user: UserProfile;
+    locale: string;
+    updateSection: (section: string) => Promise<void>;
+    actions: Actions;
+};
+
+type State = {
+    isSaving: boolean;
+    openMenu: boolean;
+    locale: string;
+    serverError?: string;
+    selectedOption: object;
+};
+
+export default class ManageLanguage extends React.PureComponent<Props, State> {
+    reactSelectContainer: React.RefObject<HTMLDivElement>;
+    constructor(props: Props) {
         super(props);
-        const locales = I18n.getLanguages();
+        const locales: any = I18n.getLanguages();
         const userLocale = props.locale;
-        const selectedOption = {value: locales[userLocale].value, label: locales[userLocale].name};
+        const selectedOption = {
+            value: locales[userLocale].value,
+            label: locales[userLocale].name
+        };
         this.reactSelectContainer = React.createRef();
 
         this.state = {
             locale: props.locale,
             selectedOption,
             isSaving: false,
-            openMenu: false,
+            openMenu: false
         };
     }
 
-    componentDidMount() {
-        if (this.reactSelectContainer.current) {
-            this.reactSelectContainer.current.addEventListener('keydown', this.handleContainerKeyDown);
+    componentDidMount(this: any) {
+        const reactSelectContainer = this.reactSelectContainer.current;
+        if (reactSelectContainer) {
+            reactSelectContainer.addEventListener(
+                'keydown',
+                this.handleContainerKeyDown
+            );
         }
     }
 
-    componentWillUnmount() {
+    componentWillUnmount(this: any) {
```

Comment:
Same here.

Reply:
getting an error as similar to the above error

Reply:
yes i am able to resolve it now.